variety/plugins/builtin/downloaders/EarthviewDownloader.py

```diff
@@ -54,27 +58,27 @@ def get_source_location(self):
         return self.ROOT_URL
 
     def fill_queue(self):
-        queue = Util.fetch_json(DATA_URL)
-        random.shuffle(queue)
-        return queue
+        random.shuffle(self.PHOTO_IDS)
+        return self.PHOTO_IDS
 
     def get_default_throttling(self):
         # throttle this source, as otherwise maps "overpower" all other types of images
         # with Variety's default settings, and we have no other way to control source "weights"
         return Throttling(max_downloads_per_hour=20, max_queue_fills_per_hour=None)
 
     def download_queue_item(self, item):
-        item = Util.fetch_json("https://earthview.withgoogle.com/_api/" + item["slug"] + ".json")
-        region = item["region"]
-        filename = "{}{} (ID-{}).jpg".format(
-            region + ", " if region and region != "-" else "", item["country"], item["id"]
-        )
-        origin_url = EarthviewDownloader.ROOT_URL + str(item["slug"])
-        image_url = item["photoUrl"]
-        if not image_url.startswith("http"):
-            image_url = "https://" + image_url
-
-        extra_metadata = {"description": item.get("name"), "author": item.get("attribution")}
-        return self.save_locally(
-            origin_url, image_url, local_filename=filename, extra_metadata=extra_metadata
+        item = Util.fetch_json(self.DATA_URL.format(photo_id=item))
+        country = item.get("geocode", {}).get("country", None)
+        filename = "{}(ID-{}).jpg".format(
+            country + " " if country else "", item["id"]
         )
+
+        # Extract image data from JSON
+        strip_prefix = lambda b: b[b.find("/9"):]
+        image_data = b64decode(strip_prefix(item["dataUri"]))
+
+        local_filepath = os.path.join(self.target_folder, filename)
+        with open(local_filepath, "wb") as f:
```

Comment:
Let's please not bypass the usage of `save_locally`, it does a lot of things other than just storing the remote image file locally.
E.g. it will create the necessary folders, if they're not already there, there's logging, metadata handling, etc.
https://github.com/varietywalls/variety/blob/36d59c573d8429ebb487e6ad51cc62a0d5e0e7a0/variety/plugins/downloaders/DefaultDownloader.py#L178-L264

Reply:
@peterlevi  the JSON is now downloaded using `save_locally`, and then opened to extract and save the image data. This still feels a little bit like a hack but its possibly the best option (?)

Please review again!